pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -385,14 +394,18 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 			},
 		},
 		Providers: providers,
-	}
+	}, nil
 }
 
 // buildLocalJwks builds local Jwks by fetching the Jwt Public Key from the URL passed if it is empty.
-func buildLocalJwks(jwksURI, jwtPubKey string) *envoy_jwt.JwtProvider_LocalJwks {
+func buildLocalJwks(jwksURI, jwtPubKey string, jwtKeyResolver *model.JwksResolver) (*envoy_jwt.JwtProvider_LocalJwks, error) {
 	if jwtPubKey == "" {
 		var err error
-		jwtPubKey, err = model.GetJwtKeyResolver().GetPublicKey(jwksURI)
+
+		if jwtKeyResolver == nil {
```

Comment:
In what case this will be nil? It seems it will never be nil, and this function should not return nil because you're just ignoring the error in `buildFilter()` and that could cause security vulnerability.

Reply:
> In what case this will be nil?

Normally this should never be nil. Just a sanity check to avoid the ref of JwtKeyResolver not being populated into the push context.

> It seems it will never be nil, and this function should not return nil because you're just ignoring the error in `buildFilter()` and that could cause security vulnerability.

That was actually my concern mentioned here. https://github.com/istio/istio/pull/30950#discussion_r579626635

Reply:
cc @lei-tang

Reply:
If it's never nil, then do not check and just use it.
We'd rather crash early instead of hiding the error which could cause real security issue if the jwt config is not properly handled. Ignoring the error with a log message in pilot will likely not be found by user, and when it is found, it is usually too late and will be a security incident because some jwt configs are ignored and requests could be bypassed without jwt authentication.

Reply:
As the errors in Istio code are propagated, the PR should follow this convention too (if there are errors, return errors).

Reply:
Thanks for the comments!
Use it directly and add comments why it should never nil.